src/group_theory/nilpotent.lean

```diff
@@ -395,3 +397,16 @@ begin
     (le_trans ((map_eq_bot_iff _).mp _) hf1)⟩,
   exact eq_bot_iff.mpr (hn ▸ (lower_central_series.map f n)),
 end
+
+lemma derived_le_lower_central (n : ℕ) : derived_series G n ≤ lower_central_series G n :=
+by { induction n with i ih, { simp }, { apply general_commutator_mono ih, simp } }
+
+theorem is_nilpotent.is_solvable (h : is_nilpotent G): is_solvable G :=
+begin
+  rw nilpotent_iff_lower_central_series at *,
+  rcases h with ⟨n, hn⟩,
```

Comment:
```suggestion
  obtain ⟨n, hn⟩ := nilpotent_iff_lower_central_series.1 h,
```

Reply:
This would leave `h` in the goal, right? I like tactics that don’t leave redundant information in the goal state…

Also, this is the same idiom as 10 line above (by @ineswright in b0e8ced14f). Should I change that too? Or just leave it is it is?

Reply:
In the case of Ines, `rw` is used on several `is_nilpotent` hypotheses. You only have one.

Reply:
Oh, fair point